File: Python/6kyu/FindTheUniqueNumber.py
from collections import Counter


def find_uniq(arr):
    # your code here
    arr.sort()

    return arr[-1] if arr[0] == arr[1] else arr[0]


def find_uniq_kata_1(arr):
    a, b = set(arr)

    return a if arr.count(a) == 1 else b


def find_uniq_kata_2(arr):
    return Counter(arr).most_common()[-1][0]


input = [0, 0, 0.55, 0, 0]
result = Counter(input).elements()
for r in result:
    print(r)

File: JavaScript/6kyu/FindTheUniqueNumber.js
function findUniq(arr) {
  var num = arr.slice(0, 3).find((n) => arr.filter((a) => a == n).length > 1);

  var i = 0;
  while (arr[i] == num) {
    i++;
  }

  return arr[i];
}

function findUniq_kata_1(arr) {
  arr.sort((a, b) => a - b);
  return arr[0] == arr[1] ? arr.pop() : arr[0];
}

function findUniq_kata_2(arr) {
  return arr.find((n) => arr.indexOf(n) === arr.lastIndexOf(n));
}

var result = findUniq_kata_2([1, 0, 0, 0, 0, 0, 0, 0]);
console.log(result);
